internal/kms/job/job.go

```diff
@@ -0,0 +1,33 @@
+package job
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/scheduler"
+)
+
+// RegisterJobs registers kms related jobs with the provided scheduler.
+func RegisterJobs(ctx context.Context, s *scheduler.Scheduler, kmsRepo *kms.Kms) error {
+	const op = "kmsjob.RegisterJobs"
+
+	dataKeyVersionDestructionMonitorJob, err := newDataKeyVersionDestructionMonitorJob(ctx, kmsRepo)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	if err := s.RegisterJob(ctx, dataKeyVersionDestructionMonitorJob); err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	for _, tableName := range kms.ListTablesSupportingRewrap() {
```

Comment:
Note: this lists the tables that are registered via `init` functions, not all tables referencing the key_id column. What guarantees that those lists are the same is a test that runs both and compares the results, which is not part of this PR.